### PR TITLE
Improve performances

### PR DESCRIPTION
**Pull request by cbellot000**
Allow disabling property evaluation by jedi
Improve performances of data upload
Improve performances of local field

### CODE REVIEW

File: ansys/dpf/core/scoping.py [hunk 1]
```diff
@@ -349,8 +353,204 @@ def deep_copy(self, server=None):
...
+    def _set_ids(self, ids):
+        """
+        Parameters
+        ----------
+        ids : list of int
+            IDs to set.
+
+        Notes
+        -----
+        Print a progress bar.
```
Comment (akaszynski) [inline ansys/dpf/core/scoping.py:454]: Does this print a progress bar?

File: ansys/dpf/core/scoping.py [hunk 2]
```diff
@@ -349,8 +353,204 @@ def deep_copy(self, server=None):
...
+    def _get_ids(self, np_array=False):
+        """
+        Returns
+        -------
+        ids : list[int], numpy.array (if np_array==True)
+            List of IDs.
+
+        Notes
+        -----
+        Print a progress bar.
```
Comment (akaszynski) [inline ansys/dpf/core/scoping.py:476]: Same here.

File: ansys/dpf/core/scoping.py [hunk 3]
```diff
@@ -349,8 +353,204 @@ def deep_copy(self, server=None):
...
+        """
+        if np_array:
+            return self._scoping_ids_copy
+        else:
```
Comment (akaszynski) [inline ansys/dpf/core/scoping.py:480]: Unnecessary else.

File: ansys/dpf/core/scoping.py [hunk 4]
```diff
@@ -349,8 +353,204 @@ def deep_copy(self, server=None):
...
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, tb):
+        if tb is None:
+            self.release_data()
+        else:
+            print(tb)
+
+    def __del__(self):
```
Comment (akaszynski) [inline ansys/dpf/core/scoping.py:546]: Consider updating the server on destruction.

File: ansys/dpf/core/misc.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 from pkgutil import iter_modules
 
 
-DEFAULT_FILE_CHUNK_SIZE = 65536
+DEFAULT_FILE_CHUNK_SIZE = 500000
```
Comment (akaszynski) [inline ansys/dpf/core/misc.py:8]: My hunch is this should be a power of 2.  For example: 2**19 or 524,288.

File: ansys/dpf/core/settings.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        import pyvista as pv
+
+        pv.OFF_SCREEN = True
+
+def set_default_pyvista_config():
+    # Configure PyVista's ``rcParams`` for dpf
+    if module_exists("pyvista"):
+        import pyvista as pv
+
+        pv.rcParams["interactive"] = True
```
Comment (akaszynski) [inline ansys/dpf/core/settings.py:22]: Consider using the new pyvista theme API.  See:

https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.theme.html?highlight=theme#pyvista.Plotter.theme

and

https://docs.pyvista.org/user-guide/themes.html?highlight=theme

and

https://docs.pyvista.org/examples/02-plot/themes.html?highlight=theme

Requires pyvista >= 0.31.0

File: ansys/dpf/core/scoping.py (hunk 1 above)
Comment (cbellot000) [reply ansys/dpf/core/scoping.py:454]: yes

File: ansys/dpf/core/scoping.py (hunk 2 above)
Comment (cbellot000) [reply ansys/dpf/core/scoping.py:476]: yes

File: ansys/dpf/core/scoping.py (hunk 4 above)
Comment (cbellot000) [reply ansys/dpf/core/scoping.py:544]: You're right, I did it

**Review by rlagha — APPROVED** [review]

**Pull request merged**